### PR TITLE
Add support for AWS China

### PR DESCRIPTION
**Pull request by anderspede**
In order to support authentication with accounts in the AWS China partition, this pull request adds partition detection based on the specified region.

### CODE REVIEW

File: assume-aws-sso-role [hunk 1]
```diff
@@ -118,6 +120,15 @@ role_info_from_profile() {
   set_account_and_role_from_role_arn
 }
 
+get_partition_from_region() {
+  if [[ ${aws_region} = cn-* ]] then
```
Comment (relausen-tlg) [inline assume-aws-sso-role:124]: I believe there is a syntax error here. There should be a `;` before `then`.

Comment (anderspede) [reply assume-aws-sso-role]: You are absolutely right and it should be fixed now.

File: assume-aws-sso-role [hunk 2]
```diff
@@ -118,6 +120,15 @@ role_info_from_profile() {
   set_account_and_role_from_role_arn
 }
 
+get_partition_from_region() {
+  if [[ ${aws_region} = cn-* ]]; then
+    aws_partition=aws-cn
```
Comment (jhd-lego) [inline assume-aws-sso-role:125]:
```suggestion
    aws_partition="aws-cn"
```

File: assume-aws-sso-role [hunk 3]
```diff
@@ -31,6 +31,7 @@ aws_role=
 aws_account=
 aws_role=
 aws_session_duration=3600
+aws_partition=aws
```
Comment (jhd-lego) [inline assume-aws-sso-role:34]:
```suggestion
aws_partition="aws"
```

**Review by jhd-lego — CHANGES_REQUESTED** [review]
Really prefer having strings be quoted.
Really nice, thanks!

File: assume-aws-sso-role (hunk 3 above)
Comment (anderspede) [reply assume-aws-sso-role]: fixed

**Comment by jhd-lego** [comment]
Awesome, thanks a lot!

**Comment by relausen-tlg** [comment]
This was a great and important PR. Thank you, Anders!

**Pull request merged**